t/mojolicious/validation_lite_app.t

```diff
@@ -39,347 +39,391 @@ any '/forgery' => sub {
+  ok !$v->has_error, 'no error';
+  ok !$v->optional('baz')->is_valid, 'not valid';
+  is_deeply $v->output, {}, 'right result';
+  ok !$v->has_error, 'no error';
+  ok !$v->optional('yada')->is_valid, 'not valid';
+  is_deeply $v->output, {}, 'right result';
+  ok !$v->has_error, 'no error';
+  ok $v->optional('bar')->is_valid, 'valid';
+  is_deeply $v->output, {bar => ['a']}, 'right result';
+  ok !$v->in('c')->is_valid, 'not valid';
+  is_deeply $v->output, {}, 'right result';
+  ok $v->has_error, 'has error';
+};
+
+subtest 'Empty string' => sub {
+  my $v = $t->app->validation->input({foo => ''});
+  ok $v->optional('foo')->is_valid, 'valid';
+  is_deeply $v->output, {foo => ''}, 'right result';
+};
+
+subtest 'Equal to' => sub {
+  my $v = $t->app->validation->input({foo => 'bar', baz => 'bar', yada => 'yada'});
+  ok $v->optional('foo')->equal_to('baz')->is_valid, 'valid';
+  is_deeply $v->output, {foo => 'bar'}, 'right result';
+  ok !$v->has_error, 'no error';
+  ok !$v->optional('baz')->equal_to('does_not_exist')->is_valid, 'not valid';
+  is_deeply $v->output, {foo => 'bar'}, 'right result';
+  ok $v->has_error, 'has error';
+  is_deeply $v->error('baz'), [qw(equal_to 1 does_not_exist)], 'right error';
+  ok !$v->optional('yada')->equal_to('foo')->is_valid, 'not valid';
+  is_deeply $v->output, {foo => 'bar'}, 'right result';
+  ok $v->has_error, 'has error';
+  is_deeply $v->error('yada'), [qw(equal_to 1 foo)], 'right error';
+  is_deeply $v->failed,        [qw(baz yada)],       'right names';
+};
+
+subtest 'In' => sub {
+  my $v = $t->app->validation->input({foo => [qw(bar whatever)], baz => [qw(yada ohoh)]});
+  ok $v->required('foo')->in(qw(23 bar whatever))->is_valid, 'valid';
+  is_deeply $v->every_param('foo'), [qw(bar whatever)], 'right results';
+  is $v->param('foo'), 'whatever', 'right result';
+  is_deeply $v->output, {foo => [qw(bar whatever)]}, 'right result';
+  ok !$v->has_error, 'no error';
+  ok !$v->required('baz')->in(qw(yada whatever))->is_valid, 'not valid';
+  is_deeply $v->output, {foo => [qw(bar whatever)]}, 'right result';
+  ok $v->has_error, 'has error';
+  is_deeply $v->error('baz'), [qw(in 1 yada whatever)], 'right error';
+  is_deeply $v->failed, ['baz'], 'right names';
+};
+
+subtest 'Like' => sub {
+  my $v = $t->app->validation->input({foo => [qw(bar whatever)], baz => [qw(yada ohoh)]});
+  ok $v->required('foo')->in(qw(23 bar whatever))->is_valid, 'valid';
+  is_deeply $v->every_param('foo'), [qw(bar whatever)], 'right results';
+  is $v->param('foo'), 'whatever', 'right result';
+  is_deeply $v->output, {foo => [qw(bar whatever)]}, 'right result';
+  ok !$v->has_error, 'no error';
+  ok !$v->required('baz')->in(qw(yada whatever))->is_valid, 'not valid';
+  is_deeply $v->output, {foo => [qw(bar whatever)]}, 'right result';
+  ok $v->has_error, 'has error';
+  is_deeply $v->error('baz'), [qw(in 1 yada whatever)], 'right error';
+  is_deeply $v->failed, ['baz'], 'right names';
```

Comment:
What is this? Why are there lots of new tests here?

Reply:
not sure what happened there. This is fixed now. Thanks.